build.project

```diff
@@ -7,22 +7,40 @@
 #
 [ "${BASH_SOURCE[0]}" -ef "$0" ] && echo "You should source this script, not execute it!" && exit 1
 
-set -e
-
+#TODO: remove
 #TARGET=armv7m4-stm32l4x6
 #TARGET=armv7m7-imxrt105x
 #TARGET=armv7m7-imxrt106x
 #TARGET=armv7m7-imxrt117x
 #TARGET=armv7a7-imx6ull
 #TARGET=ia32-generic
 #TARGET=riscv64-spike
-TARGET=riscv64-virt
+#TARGET=riscv64-virt
+
+# optional args will be printed in red
+b_list_available_targets() {
+    [ "$#" -gt 0 ] && echo -e "\e[1;31m$*\e[0m"
+    echo -e "Please specify ia valid traget by setting TARGET variable to one of:"
+    ls -1 _targets/build.project.* | cut -d. -f 3
+}
+
+# it's best to provide TARGET by env variable
+if [ -z "$TARGET" ]; then
+    b_list_available_targets "TARGET variable not set"
+    exit 1
+fi
+
 
+TARGET_FAMILY=$(echo "$TARGET" | awk -F- '{ print $1 }')
+TARGET_SUBFAMILY="$TARGET_FAMILY-$(echo "$TARGET" | awk -F- '{ print $2 }')"
```

Comment:
This can be simpliefied to:
```
TARGET_FAMILY=$(echo "$TARGET" | cut -d- -f1)
TARGET_SUBFAMILY=$(echo "$TARGET" | awk -F- '{ print $1"-"$2 }')
```
or without creating processes:
```
TARGET_FAMILY=${TARGET%%-*}
tmp=${TARGET#*-}
TARGET_SUBFAMILY="$TARGET_FAMILY-${tmp%%-*}"
unset tmp
```

Reply:
Yeah, I wanted to do something with that ugly one-liner but ended with using `printf` in `awk` - which was not better by any means.

I think we can use string substitution as You suggested (as it's POSIX sh compliant), but I would prefer to do it separately from this PR (probably introduce it as a function to `phoenix-rtos-build/build.subr`?).

For now I've used Your version of `awk` script as it's tidier than the previous solution.